pvlib/tests/test_spectrum.py

```diff
@@ -123,24 +123,70 @@ def test_get_example_spectral_response():
     assert_allclose(sr, expected, rtol=1e-5)
 
 
+@fail_on_pvlib_version('0.12')
 def test_get_am15g():
     # test that the reference spectrum is read and interpolated correctly
     e = spectrum.get_am15g()
     assert_equal(len(e), 2002)
     assert_equal(np.sum(e.index), 2761442)
     assert_approx_equal(np.sum(e), 1002.88, significant=6)
 
-    wavelength = [270, 850, 950, 1200, 4001]
-    expected = [0.0, 0.893720, 0.147260, 0.448250, 0.0]
+    wavelength = [270, 850, 950, 1200, 1201.25, 4001]
+    expected = [0.0, 0.893720, 0.147260, 0.448250, 0.4371025, 0.0]
 
     e = spectrum.get_am15g(wavelength)
```

Comment:
We should wrap all calls to `get_am15g` with a context: `with pytest.warns(pvlibDeprecationWarning, match="get_standard_spectrum instead"):`

This accomplishes two things: check that the warning is being emitted as intended, and prevent the warning from cluttering the test output logging.